include/swift/AST/Evaluator.h

```diff
@@ -283,7 +283,7 @@ class Evaluator {
   typename Request::OutputType getResultUncached(const Request &request) {
     // Clear out the dependencies on this request; we're going to recompute
     // them now anyway.
-    dependencies[request].clear();
+    dependencies[AnyRequest(request)].clear();
```

Comment:
Is there a `find_as`-like trick we should do here?

Reply:
I tried one, but it turns out this is doing double-duty in clearing existing dependencies and in inserting an entry into the table, so I left it doing the allocation. Arguably we could split that out into two operations, but I don't think we throw away cached results that often anyway.

Reply:
Most requests are cached, so this probably doesn't happen too often. Worst case, if it becomes a problem later, we can extend `DenseMap` with API for this.